grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/atom/binary/RelationshipAtom.java

```diff
@@ -539,32 +539,27 @@ else if(compatibleTypesFromRoles.isEmpty() || types.isEmpty()){
      * {@link EntityType}s only play the explicitly defined {@link Role}s (not the relevant part of the hierarchy of the specified {@link Role}) and the {@link Role} inherited from parent
      * @return list of {@link RelationshipType}s this atom can have ordered by the number of compatible {@link Role}s
      */
-    public ImmutableList<Type> inferPossibleTypes(Answer sub) {
+    public ImmutableList<Type> inferPossibleTypes() {
```

Comment:
I tried to remove this and stuff started breaking, so @kasper-piskorski suggested I roll it back. But apparently it's working for you soooo

Reply:
Well in general things shouldn't break as this deals with type inference. It's just that less will be inferred. I will double check. Probably on the finance dataset.

Reply:
`Answer sub` was never used anywhere in the chain so that's why I removed it and then when you do that the whole method looks alot like the existing one.